sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -129,14 +132,39 @@ export class StudentControllerService {
       "Content-Disposition",
       `attachment; filename=${studentFile.fileName}`,
     );
-    response.setHeader("Content-Type", studentFile.mimeType);
-    response.setHeader("Content-Length", studentFile.fileContent.length);
 
-    const stream = new Readable();
-    stream.push(studentFile.fileContent);
-    stream.push(null);
-
-    stream.pipe(response);
+    // Temporary code to be changed in the upcoming effort
+    // when the the files will no longer be saved on DB.
+    const stopwatchLabel = `Downloaded file ${uniqueFileName}`;
+    console.time(stopwatchLabel);
+    try {
+      const fileContent = await this.objectStorageService.getObject(
+        uniqueFileName,
+      );
+      console.info("Downloaded file using S3 storage.");
+      // Populate file information received from S3 storage.
+      response.setHeader("Content-Type", fileContent.contentType);
+      response.setHeader("Content-Length", fileContent.contentLength);
+      fileContent.body.pipe(response);
+    } catch (error: unknown) {
+      if (error instanceof NoSuchKey) {
+        console.info("File not present on S3 storage.");
+      } else {
+        console.error(parseJSONError(error));
+      }
+      // Fallback to use the DB file in case the object storage failed
+      // or the file or never uploaded to S3.
+      // Populate fil information from DB.
+      console.info("Retrieving from DB.");
+      response.setHeader("Content-Type", studentFile.mimeType);
+      response.setHeader("Content-Length", studentFile.fileContent.length);
+      const stream = new Readable();
+      stream.push(studentFile.fileContent);
+      stream.push(null);
+      stream.pipe(response);
+    } finally {
+      console.timeEnd(stopwatchLabel);
```

Comment:
For S3 operation time why didn't we just cover the line 141 alone??

Reply:
The idea was just to try to get the difference between the overall operation. The same file can be uploaded twice and one can be deleted from the storage which would allow both times to be tested.
In the end, I believe that either way has the potential to give us the data we need.